Fix C++ build issues w/ LLVM 14 (C++17 support)

andreittr opened the pull request:
A set of fixes to enable building alongside LLVM 14 libcxx & friends:
- wrong macro set in build options
- missing declaration for prctl

This is part of a patch set consisting of:
 - [libcxx](https://github.com/unikraft/lib-libcxx/pull/28)
 - [libcxxabi](https://github.com/unikraft/lib-libcxxabi/pull/4)
 - [lib-compiler-rt](https://github.com/unikraft/lib-compiler-rt/pull/12)
 - [libunwind](https://github.com/unikraft/lib-libunwind/pull/7)
 - [musl](https://github.com/unikraft/lib-musl/pull/45)

Tested with GCC 11, 12 & 13, and clang 16.

edit: rebased & removed already upstreamed patches
edit2: rebased

andreittr committed:
Fix typo in macro definition

Previously we defined the macro `UK_LIBC_SYSCALL`, however this is not
used anywhere in unikraft; `UK_LIBC_SYSCALLS` is most likely what was
meant. This commit fixes the typo.

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

andreittr committed:
Include musl's own headers when building

This fixes a build error where an internal use of `prctl` fails due to a
missing prototype declaration.

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

rares-miculescu reviewed (approved):
It works for me. Thanks, @andreittr!

Reviewed-by: Rares Miculescu <miculescur@gmail.com>

StefanJum reviewed (changes requested):
@andreittr the [ioctl signature commit](https://github.com/unikraft/lib-musl/pull/45/commits/ae4b8bf7526655a2dfa86b7e3089eee491eb2835) is already covered by [this](https://github.com/unikraft/lib-musl/commit/cb8f2dab68eb1ad5b1604617c52077557125210c), please remove that one.

andreittr commented:
> @andreittr the [ioctl signature commit](https://github.com/unikraft/lib-musl/pull/45/commits/ae4b8bf7526655a2dfa86b7e3089eee491eb2835) is already covered by [this](https://github.com/unikraft/lib-musl/commit/cb8f2dab68eb1ad5b1604617c52077557125210c), please remove that one.

Thanks for the heads-up, removed and rebased.

StefanJum reviewed (approved):
All good, thanks.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand reviewed (approved):
I tested this as well, and it works as expected. Thanks, @andreittr.

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>

eduardvintila reviewed (approved):
Thanks, everyone!

Approved-by: Eduard Vintilă <eduard.vintila47@gmail.com>
